projects/step-frontend/src/lib/modules/timeseries/performance-view/filter-bar/filter-bar.component.ts

```diff
@@ -1,68 +1,137 @@
-  readonly EMIT_DEBOUNCE_TIME = 300;
   @Input() context!: TimeSeriesContext;
+  @Input() performanceViewSettings!: PerformanceViewSettings;
+  @Input() defaultFilterOptions: TsFilterItem[] = [];
+
   @Output() onFiltersChange = new EventEmitter<TsFilterItem[]>();
+  @Output() onGroupingChange = new EventEmitter<string[]>();
+
+  @ViewChild(PerformanceViewTimeSelectionComponent) timeSelection?: PerformanceViewTimeSelectionComponent;
+
+  @ViewChildren(FilterBarItemComponent) filterComponents?: QueryList<FilterBarItemComponent>;
+  @ViewChildren('appliedFilter', { read: ElementRef }) appliedFilters?: QueryList<ElementRef<HTMLElement>>;
+
   private emitFilterChange$ = new Subject<void>();
 
-  items: TsFilterItem[] = [];
+  readonly EMIT_DEBOUNCE_TIME = 300;
+  readonly FilterBarItemType = FilterBarItemType;
 
-  @Input() defaultFilterOptions: TsFilterItem[] = [];
+  filters: TsFilterItem[] = [];
+  moreFilters: TsFilterItem[] = [];
+
+  constructor(private _changeDetectorRef: ChangeDetectorRef) {}
 
   ngOnInit(): void {
     if (!this.context) {
       throw new Error('Context input is mandatory');
     }
+
     this.emitFilterChange$.pipe(debounceTime(this.EMIT_DEBOUNCE_TIME)).subscribe(() => {
-      this.onFiltersChange.emit(this.items.filter(FilterUtils.filterItemIsValid)); // we ignore empty filters
+      // we ignore empty filters
+      this.onFiltersChange.emit(this.filters.filter(FilterUtils.filterItemIsValid));
     });
   }
 
-  handleFilterChange(item: TsFilterItem) {
+  ngOnDestroy(): void {
+    this.emitFilterChange$.complete();
+    this.onFiltersChange.complete();
+  }
+
+  handleFilterChange() {
     this.emitFilterChange$.next();
   }
 
   addFilterItem(item: TsFilterItem) {
-    let filterExists = this.items.find((i) => i.attributeName === item.attributeName);
-    if (filterExists) {
-      return;
+    const filterIndex = this.filters.findIndex((i) => i.attributeName === item.attributeName);
+
+    if (filterIndex !== -1) {
+      const index =
+        filterIndex < this.filters.length - this.moreFilters.length
+          ? filterIndex
+          : filterIndex + this.moreFilters.length;
+
+      this.filterComponents!.toArray()[index].menuTrigger!.openMenu();
     } else {
-      this.items.push(JSON.parse(JSON.stringify(item))); // make a clone
+      this.addFilter(JSON.parse(JSON.stringify(item)));
     }
   }
 
   addCustomFilter(type: FilterBarItemType) {
-    this.items.push({
+    this.addFilter({
       attributeName: '',
       type: type,
       label: '',
     });
   }
 
   removeFilterItem(index: number) {
-    let itemToDelete = this.items[index];
-    this.items.splice(index, 1);
+    let itemToDelete = this.filters[index];
+
+    this.filters.splice(index, 1);
+
     if (FilterUtils.filterItemIsValid(itemToDelete)) {
       this.emitFilterChange$.next();
     }
   }
 
-  get FilterBarItemType() {
-    return FilterBarItemType;
+  initMoreFilters(): void {
```

Comment:
so basically there are 2 containers to hold the filters, that are interconnected; the filters are held in the filters and moreFilters arrays, and rendered in the div.applied-filters and div.more-applied-filters, respectively; the div.applied-filters has ```overflow: hidden;```, ```flex-wrap: wrap;``` and ```height: 48px;``` (the height of a filter), so that we see only one row of filters while the rest are wrapped onto multiple lines below; whenever the filter bar container is resized, we go though all the filter html elements in the div.applied-filters, take their ```top``` coordinates, order them, and then filter them out so that we determine which ones are wrapped and hidden by the overflow; then we put them in the div.more-applied-filters

Reply:
I get that "moreFilters" might not be the ideal name but couldn't find one. Would say we rename it when a better name comes up.